src/pages/dashboard/index.js

```diff
@@ -10,25 +10,34 @@ import "./style.css";
 import useAuth from "../../hooks/useAuth";
 
 const Dashboard = () => {
+  const [searchVal, setSearchVal] = useState("");
+  const [searchResults, setSearchResults] = useState([]); // State for search results
+  const [isLoading, setIsLoading] = useState(false); // Loading state
+  const [errorMessage, setErrorMessage] = useState(""); // Error message state
+
 	const [searchVal, setSearchVal] = useState('');
 	 const { user } = useAuth();
 
-	const onChange = (e) => {
-		setSearchVal(e.target.value);
-	};
 
-	// Use the useModal hook to get the openModal and setModal functions
-	const { openModal, setModal } = useModal();
+  const onChange = (e) => {
+    setSearchVal(e.target.value);
+  };
+
+  // Use the useModal hook to get the openModal and setModal functions
+  const { openModal, setModal } = useModal();
 
-	// Create a function to run on user interaction
-	const showModal = () => {
-		// Use setModal to set the header of the modal and the component the modal should render
-		setModal("Create a post", <CreatePostModal />); // CreatePostModal is just a standard React component, nothing special
+  // Create a function to run on user interaction
+  const showModal = () => {
```

Comment:
what modal are we showing? `showCreatePostModal` would be more appropriate as a name for the function